[DEVX-1677] Add docs for saucectl json reporter (#1214)
saucelabs/sauce-docs: docs/mobile-apps/automated-testing/espresso-xcuitest/espresso.md

    filename: saucectl-report.xml
```

---
### `json`
<p><small>| OPTIONAL | OBJECT |</small></p>

The JSON reporter gathers test results from all jobs in the JSON format into a single report

```yaml
reporters:

>   junit:

Reviewer: Certainly not :D

    enabled: true
    filename: saucectl-report.json

> ```

Reviewer: The webhook should also show up here.

---
#### `enabled`
<p><small>| OPTIONAL | BOOLEAN |</small></p>

Toggles the reporter on/off.

```yaml
    enabled: true
```

---
#### `webhookURL`
<p><small>| OPTIONAL | STRING |</small></p>

> Specifies the webhook URL. When saucectl test is finished, it'll send a HTTP POST payload to the configured webhook URL.

Reviewer: I think we need to document what the payload looks like.
```suggestion
Specifies the webhook URL. When saucectl test is finished, it'll send an HTTP POST with a JSON payload to the configured webhook URL.
```

    filename: saucectl-report.xml
```

---
### `json`
<p><small>| OPTIONAL | OBJECT |</small></p>

> The JSON reporter gathers test results from all jobs and writes into a JSON format report.

Reviewer: A bit more fluent:
```suggestion
The JSON reporter gathers test results from all jobs and combines them into a single report.
```

```yaml
reporters:
  json:
    enabled: true
    filename: saucectl-report.json
    webhookURL: https://my-webhook-url
```

---

> #### `enabled`
> <p><small>| OPTIONAL | BOOLEAN |</small></p>
> 
> Toggles the reporter on/off.
> 
> ```yaml
>     enabled: true
> ```
> 
> ---
> #### `webhookURL`
> <p><small>| OPTIONAL | STRING |</small></p>
> 
> Specifies the webhook URL. When saucectl test is finished, it'll send an HTTP POST with a JSON payload to the configured webhook URL.
> 
> ```yaml
>     webhookURL: https://my-webhook-url
> ```
> 
> ---
> #### `filename`

Reviewer 2: Per our style, these all need to be H3 so that they show up in the right-hand nav.

```suggestion
### `enabled`
<p><small>| OPTIONAL | BOOLEAN |</small></p>

Toggles the reporter on/off.

```yaml
    enabled: true
```

---
### `webhookURL`
<p><small>| OPTIONAL | STRING |</small></p>

Specifies the webhook URL. When saucectl test is finished, it'll send an HTTP POST with a JSON payload to the configured webhook URL.

```yaml
    webhookURL: https://my-webhook-url
```

---
### `filename`
```

Author: Hey @kimsauce , I'm just wondering how to add these fields with the inheriting orders. `enabled`, `webhookURL` and `filename` are all under `json` fields. If I set them all as H3, will it break the inheriting orders in the nav?

Reviewer 2: I see what you mean. I'm going to pull this branch locally to reformat and then send back to you guys for review.

<p><small>| OPTIONAL | STRING |</small></p>

> Specifies the report filename. When saucectl test is finished, it'll log the test result in this file. If the field is not set, the default one is saucectl-report.json

Reviewer: A bit more concise without any loss of information:
```suggestion
Specifies the report filename. Defaults to "saucectl-report.json".
```
